multipy/runtime/interpreter/plugin_registry.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <ATen/core/ivalue.h>
+#include <pybind11/embed.h>
+#include <pybind11/functional.h>
+
+#include <multipy/runtime/interpreter/Optional.hpp>
+
+namespace py = pybind11;
+
+namespace multipy {
+
+class Converter {
+ public:
+  virtual ~Converter() = default;
+
+  virtual multipy::optional<at::IValue> toTypeInferredIValue(
```

Comment:
Given we are using C++17 we should just be able to use https://en.cppreference.com/w/cpp/utility/optional (could also do at a future date when cleaning up the library).

Reply:
For now I think keeping it all consistent with `multipy::optional` is best -- C++14 support might be useful for folks